File: HolyChildhood/Models/Page.cs
﻿using System.Collections.Generic;

namespace HolyChildhood.Models
{
    public class Page
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public List<PageContent> PageContents { get; set; }
        public Page Parent { get; set; }
        public List<Page> Children { get; set; }
    }
}

File: HolyChildhood/ClientApp/src/app/shared/models/page.ts
export interface Page {
    id: number;
    title: string;
    children: Array<Page>;
    parent: Page;
    pageContents: PageContent[];
}

export interface PageContent {
    id: number;
    content: string;
    contentType: string;
    editing: boolean;
    x: number;
    y: number;
    height: number;
    width: number;
    page: Page;
}
